lib/msal-core/src/ScopeSet.ts

```diff
@@ -1,69 +1,189 @@
+     * @param otherScopes
+     */
+    intersectingScopeArrays(otherScopes: Array<string>): boolean {
+        return this.unionScopeSets(otherScopes).size < this.scopes.size + otherScopes.length;
+    }
+
+    /**
+     * 
+     * @param otherScopes 
+     */
+    unionScopeSets(otherScopes: Array<string>): Set<string> {
+        return new Set<string>([ ...StringUtils.convertArrayEntriesToLowerCase(otherScopes), ...Array.from(this.scopes) ]);
     }
 
     /**
-   * toLower
-   *
-   * @param scopes
-   */
-    // TODO: Rename this, too generic name for a function that only deals with scopes
-    static convertToLowerCase(scopes: Array<string>): Array<string> {
-        return scopes.map(scope => scope.toLowerCase());
+     * Check if a given scope is present in the request
+     *
+     * @param cachedScopes
+     * @param scopes
+     */
+    containsScope(scope: string): boolean {
+        return this.scopes.has(scope);
     }
 
     /**
-   * remove one element from a scope array
-   *
-   * @param scopes
-   * @param scope
-   */
-    // TODO: Rename this, too generic name for a function that only deals with scopes
-    static removeElement(scopes: Array<string>, scope: string): Array<string> {
-        return scopes.filter(value => value !== scope);
+     * @ignore
+     * Appends extraScopesToConsent if passed
+     * @param {@link AuthenticationParameters}
+     */
+    appendExtraScope(newScope: string): void {
+        this.scopes.add(newScope);
+    }
+
+    /**
+     * @ignore
+     * Appends extraScopesToConsent if passed
+     * @param {@link AuthenticationParameters}
+     */
+    appendExtraScopes(newScopes: Array<string>): void {
+        this.validateInputScopes(newScopes);
+        this.scopes = this.unionScopeSets(newScopes);
+    }
+
+    /**
+     * remove one element from set of scopes
+     *
+     * @param scope
+     */
+    removeElement(scope: string): void {
+        this.scopes.delete(scope);
+    }
+
+    /**
+     * @hidden
+     * @ignore
+     *
+     * ScopeSet helper to 
+     * @param matchingAccount boolean telling whether account object matches value in cache
+     * @param silentCall boolean telling whether or not silent call
+     *
+     * @returns {string} token type: id_token or access_token
+     *
+     */
+    getTokenType(matchingAccount: boolean, silentCall: boolean): string {
+        // if account is passed and matches the account object/or set to getAccount() from cache
+        // if client-id is passed as scope, get id_token else token/id_token_token (in case no session exists)
+        if (!matchingAccount) {
+            return silentCall && this.isLoginCall() ? ResponseTypes.id_token : ResponseTypes.id_token_token;           
+        } else {
+            return this.isLoginCall ? ResponseTypes.id_token : ResponseTypes.token;
+        }
+    }
+
+    /**
+     * ScopeSet helper which will tell whether set of scopes is being used for login calls.
+     */
+    isLoginCall(): boolean {
```

Comment:
name, scopeset is not tied to a call, its just constructed and used however it may used. 
containsClientScopeOnly?